pkg/secret/secret.go

```diff
@@ -0,0 +1,186 @@
+package secret
+
+import (
+	"fmt"
+	"github.com/appuio/seiso/pkg/kubernetes"
+	"github.com/appuio/seiso/pkg/openshift"
+	"github.com/appuio/seiso/pkg/util"
+	log "github.com/sirupsen/logrus"
+	"github.com/thoas/go-funk"
+	v1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	core "k8s.io/client-go/kubernetes/typed/core/v1"
+	"sort"
+	"time"
+)
+
+type (
+	Service interface {
+		ListNamesAndLabels() (resourceNames, labels []string, err error)
+		List(listOptions metav1.ListOptions) (resources []v1.Secret, err error)
+		GetUnused(namespace string, secrets []v1.Secret) (unusedSecrets []v1.Secret, funcErr error)
+		Delete(secrets []v1.Secret)
+		FilterByTime(secrets []v1.Secret, olderThan time.Time) (filteredSecrets []v1.Secret)
+		FilterByMaxCount(secrets []v1.Secret, keep int) (filteredSecrets []v1.Secret)
+		Print(secrets []v1.Secret)
+	}
+
+	SecretsService struct {
+		configuration Configuration
+		client        core.SecretInterface
+		helper        kubernetes.Kubernetes
+	}
+)
+
+type Configuration struct {
+	Batch bool
+}
+
+// NewSecretsService creates a new Service instance
+func NewSecretsService(client core.SecretInterface, helper kubernetes.Kubernetes, configuration Configuration) Service {
+	return SecretsService{
+		client:        client,
+		helper:        helper,
+		configuration: configuration,
+	}
+}
+
+// ListNamesAndLabels return the names and labels of all secrets
+func (ss SecretsService) ListNamesAndLabels() (resourceNames, labels []string, err error) {
+	secrets, err := ss.List(metav1.ListOptions{})
```

Comment:
The `err` is never returned. This is a bug:
```
$ seiso secrets   
INFO Please use labels to select Secrets. The following Secrets and Labels are available:  
 - namespace=<redacted> 
 - 🎫 labels="[]" 
 - 🔐 secrets="[]"

$ oc get secrets
error: You must be logged in to the server (Unauthorized)
```

Reply:
fixed